src/main/kotlin/no/skatteetaten/aurora/boober/feature/HerkimerVaultFeature.kt

```diff
@@ -0,0 +1,192 @@
+        val values = jsonMapper().convertValue<Map<String, String>>(response.claims.first().credentials)
+        return newSecret {
+            metadata {
+                name = response.name.ensureStartWith(adc.name, "-")
+                    .ensureEndsWith(FEATURE_FIELD, "-")
+                    .toLowerCase().replace("_", "-")
+                namespace = adc.namespace
+            }
+            data = values.mapValues { content -> Base64.encodeBase64String(content.value.toByteArray()) }
+        }
+    }
+
+    override fun modify(adc: AuroraDeploymentSpec, resources: Set<AuroraResource>, context: FeatureContext) {
+        val envVars = context.herkimerVaultCredentials.flatMap { (vaultCredential, secrets) ->
+            if (vaultCredential.multiple) {
+                secrets.generateEnvVarsForMultipleValues(vaultCredential.prefix)
+            } else {
+                secrets.generateEnvVarsForSingleValue(vaultCredential.prefix)
+            }
+        }
+
+        resources.addEnvVarsToMainContainers(envVars, javaClass)
+    }
+
+    private fun List<Secret>.generateEnvVarsForSingleValue(prefix: String): List<EnvVar> =
+        this.first().createEnvVarRefs(prefix = "${prefix}_")
+
+    private fun List<Secret>.generateEnvVarsForMultipleValues(prefix: String): List<EnvVar> =
+        this.mapIndexed { index, secret -> secret.createEnvVarRefs(prefix = "${prefix}_${index}_") }.flatten()
+
+    private fun AuroraDeploymentSpec.findAllConfiguredCredentials() =
+        this.getSubKeyValues(FEATURE_FIELD).mapNotNull {
+            this.findConfiguredCredential(it)
+        }
+
+    private fun AuroraDeploymentSpec.findConfiguredCredential(credentialKey: String): HerkimerVaultCredential? {
+        val enabled: Boolean = this["$FEATURE_FIELD/$credentialKey/enabled"]
+        if (!enabled) return null
+        return HerkimerVaultCredential(
+            key = credentialKey,
+            prefix = this.getOrNull<String>("$FEATURE_FIELD/$credentialKey/prefix") ?: credentialKey,
+            resourceKind = this["$FEATURE_FIELD/$credentialKey/resourceKind"],
+            multiple = this["$FEATURE_FIELD/$credentialKey/multiple"]
+
+        )
+    }
+
+    private fun List<HerkimerVaultCredential>.validateSharedPrefixConstraints(): List<IllegalArgumentException> =
+        this.groupBy { it.prefix }
+            .mapNotNull { (prefix, vaultCredentials) ->
+
+                val credentialsConfiguredAsMultiple = vaultCredentials.filter { it.multiple }
+                val credentialsConfiguredAsSingle = vaultCredentials.filter { !it.multiple }
+
+                if (credentialsConfiguredAsMultiple.isNotEmpty() && credentialsConfiguredAsSingle.isNotEmpty()) {
+                    IllegalArgumentException(
+                        """
+                        |The shared prefix=$prefix has been configured with both multiple=false and multiple=true.
```

Comment:
Forandret denne, hva tenker du om denne nå @bjartek ?

Reply:
Kan man bruke partition fremfor de to filtrene der? Det er en liten detail da.

Reply:
Feilmeldingen er bra synes jeg, men den inneholder linjebrekk og det gjør ingen av de andre feilmeldingene vi har, jeg vet ikke helt hvordan det vil slå ut.